front/src/Xmpp/MucRoom.ts

```diff
@@ -0,0 +1,215 @@
+import type { RoomConnection } from "../Connexion/RoomConnection";
+import xml from "@xmpp/xml";
+import jid, { JID } from "@xmpp/jid";
+import { gameManager } from "../Phaser/Game/GameManager";
+import type { Readable, Writable } from "svelte/store";
+import { writable } from "svelte/store";
+import ElementExt from "./Lib/ElementExt";
+import { getRoomId } from "../Stores/GuestMenuStore";
+import { numberPresenceUserStore } from "../Stores/MucRoomsStore";
+import { v4 as uuidv4 } from "uuid";
+import {localUserStore} from "../Connexion/LocalUserStore";
+
+export const USER_STATUS_AVAILABLE = "available";
+export const USER_STATUS_DISCONNECTED = "disconnected";
+export type User = {
+    nick: string,
+    roomId: string;
+    status: string;
+    isInSameMap: boolean;
+};
+export type UserList = Map<string, User>;
+export type UsersStore = Readable<UserList>;
+
+export function goToWorkAdventureRoomId(roomId: string, mouseEvent: MouseEvent | undefined) {
```

Comment:
Add loader when user clic on the button. Clear the loader when user have message from the back and is negative to move.